wsproxy/proxy_test.go

```diff
@@ -1137,31 +1095,25 @@ func TestProxyWebSocketPath(t *testing.T) {
 	server := httptest.NewServer(proxy)
 	defer server.Close()
 
-	configurer := func() (client.Config, error) {
-		return client.Config{
-			ID:        "workerid",
-			Token:     workeridjwt,
-			ProxyAddr: "ws://register.tcproxy.dev:" + getPort(server.URL),
-		}, nil
-	}
-
-	client, err := client.New(configurer)
+	clientLogger := genLogger()
+	wsURL := util.MakeWsURL(server.URL)
+	client, err := client.New(testConfigurer("myclient", wsURL, client.RetryConfig{}, clientLogger))
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	reqURI := "/hookid/path?foo=bar&abc=def"
-	uri := "ws://workerid.tcproxy.dev:" + getPort(server.URL) + reqURI
+	uri := "ws://localhost:" + getPort(server.URL) + "/myclient" + reqURI
 	clientHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if !websocket.IsWebSocketUpgrade(r) {
-			t.Fatal("request should be a websocket upgrade")
+			panic("request should be a websocket upgrade")
```

Comment:
Why change it to `panic`? It prevents independent tests to execute.

Reply:
You can't call t.Fatal in a goroutine, and this func runs in a goroutine.  When this test failed, it didn't give any error output.  I suppose the real solution is to connect a `chan` back to the main test goroutine and call `t.Fatal` there, but that's a lot of work.  This succeeds in giving a useful message on failure, and works fine in the success case.

Reply:
(the linter picked up some examples of this a few PR's ago, but can't statically determine which funcs are called in other goroutines, so I only noticed this one when it failed without useful output)